File: Nim/Hanoi/hanoi.nim
from strutils import parseInt

proc hanoi(n: int, froms: string, to: string, via: string) =
  if n > 1:
    hanoi(n - 1, froms, via, to)
    echo froms, " -> ", to
    hanoi(n - 1, via, to, froms)
  else:
    echo froms, " -> ", to

var n = parseInt(readLine(stdin))
hanoi(n, "a", "b", "c")

File: Perl/Hanoi/hanoi.pl
sub hanoi {
    my ($n, $from, $to, $via) = @_;
    if ($n > 1) {
        &hanoi($n - 1, $from, $via, $to);
        print $from . " -> " . $to . "\n";
        &hanoi($n - 1, $via, $to, $from);
    } else {
        print $from . " -> " . $to . "\n";
    }
}

my $n = <STDIN>;
&hanoi($n, "a", "b", "c");
